catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectExporter.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.io.asynctask
+
+import android.content.Context
+import android.util.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.catrobat.catroid.R
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.io.StorageOperations
+import org.catrobat.catroid.io.ZipArchiver
+import org.catrobat.catroid.utils.notifications.NotificationData
+import org.catrobat.catroid.utils.notifications.StatusBarNotificationManager
+import java.io.File
+import java.io.IOException
+import java.lang.ref.WeakReference
+
+class ProjectExporter(
+    private val projectDir: File,
+    private val notificationData: NotificationData,
+    context: Context
+) {
+    private val contextWeakReference: WeakReference<Context> = WeakReference(context)
+
+    @JvmOverloads
+    fun exportProjectToExternalStorageAsync(
+        scope: CoroutineScope = CoroutineScope(Dispatchers.IO)
+    ) {
+        scope.launch {
+            exportProjectToExternalStorage()
+        }
+    }
+
+    fun exportProjectToExternalStorage() {
+        deleteUndoFile(projectDir)
+        val projectZip = File(
+            Constants.EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY,
+            projectDir.name + Constants.CATROBAT_EXTENSION
+        )
```

Comment:
Please move this to a method and dependency inject the export directory. 
(=> no deletion of the export directory is necessary if a test zip can be injected)